internal/configs/virtualserver.go

```diff
@@ -1329,7 +1338,11 @@ func upstreamHasKeepalive(upstream conf_v1.Upstream, cfgParams *ConfigParams) bo
 	return cfgParams.Keepalive != 0
 }
 
-func generateRewrites(path string, proxy *conf_v1.ActionProxy, internal bool, originalPath string) []string {
+func generateRewrites(path string, proxy *conf_v1.ActionProxy, internal bool, originalPath string, grpcEnabled bool) []string {
+	if grpcEnabled && internal {
```

Comment:
this is still pending -- https://github.com/nginxinc/kubernetes-ingress/pull/2110#discussion_r741533516

basically, when `grpcEnabled && internal` is `true`, we can still have `proxy != nil && proxy.RewritePath != ""` (meaning there is rewrite configured using the proxy action). In this case we should not return, but rather add `^ $request_uri` (without the break) as a first rewrite in the result.

Reply:
Oh I must have missed this one! Thanks 😄 

I've updated that function and added another test for the case where `grpcEnabled && internal` is `true`,and we still have `proxy != nil && proxy.RewritePath != ""`